src/app/mat-select-search/mat-select-search.component.ts

```diff
@@ -269,8 +295,13 @@ export class MatSelectSearchComponent implements OnInit, OnDestroy, ControlValue
     private _viewportRuler: ViewportRuler,
     @Optional() @Inject(MatOption) public matOption: MatOption = null,
     private liveAnnouncer: LiveAnnouncer,
-    @Optional() @Inject(MatFormField) public matFormField: MatFormField = null
+    @Optional() @Inject(MatFormField) public matFormField: MatFormField = null,
+    @Optional() @Inject(MATSELECTSEARCH_GLOBAL_OPTIONS) globalOptions?: MatSelectSearchOptions
   ) {
+    if (globalOptions) {
```

Comment:
we should rather explicitly assign the properties (in a separate function)

Reply:
Sure I could do that - but isn't it a bit redundant? Are you afraid one could put there properties not needed / override non overridable? I'd prefer the current solution instead of many of lines. But up to you! Your lib, your rules.

Reply:
Yes, it would allow to override anything, maybe even inadvertently.
So I would prefer explicity assignments (can be a for loop of a list of properties, typed as keyof MatSelectSearchComponent, for compiler safety

Reply:
Saw you mentioned it already. Give it this last fix and merge it! 🥳